### PR TITLE
[Main Page] Embedded Video + Navbar

### PR DESCRIPTION
**Pull request by aaneres**
## Summary
- This PR is a sacrificial PR where I added a bunch of necessary custom components to layout the foundations before integrating components together. Those components include: Sidebar, Pagination, Header.
- Some of the focused feature:
  - Different sidebar depending if it's a normal page or a doc page
  - Isolated docs (a.k.a, Lab Notebook, ELSI, Idreas are no longer combined together like the default). This also involves the pagination as well!!
 - THOUGH, it is still not **responsive**. There should be a separate PR later to test + implement responsiveness of the page.
## Demo Video
[image](https://www.youtube.com/watch?v=5xYJSD8eKDY)

### CODE REVIEW

File: src/content/docs/index.mdx [hunk 1]
```diff
@@ -1,36 +1,10 @@
...
-  image:
-    file: ../../assets/houston.webp
-  actions:
-    - text: Example Guide
-      link: /guides/example/
-      icon: right-arrow
-    - text: Read the Starlight docs
-      link: https://starlight.astro.build
-      icon: external
-      variant: minimal
```
Comment (IcedTea2K) [inline src/content/docs/index.mdx]: This part is necessary for the main page.

File: src/styles/navbar.css [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+/* Mobile styles */
+@media (max-width: 768px) {
+  .nav-links {
+    display: none; /* Hide links */
+  }
+}
+
+
```
Comment (IcedTea2K) [inline src/styles/navbar.css]: will need to convert this to `tailwindcss` for consistency with other components/pages

File: src/components/starlight/Header.astro [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      >
+        <path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path>
+        <polyline points="15 3 21 3 21 9"></polyline>
+        <line x1="10" y1="14" x2="21" y2="3"></line>
+      </svg>
+    </a>
+  </div>
+</nav>
+
+<style>
```
Comment (IcedTea2K) [inline src/components/starlight/Header.astro]: Will have to convert this entire blcok to tailwindcss

File: src/components/YouTubeVideo.astro [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    height="315"
+    src={`https://www.youtube.com/embed/${videoId}`}
+    title="YouTube video player"
+    frameborder="0"
+    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+    allowfullscreen
+  ></iframe>
+</div>
+
+<style>
```
Comment (IcedTea2K) [inline src/components/YouTubeVideo.astro]: convert to tailwindcss

**Review by IcedTea2K — CHANGES_REQUESTED** [review]
Good stuff. We just need to convert the css files to tailwindcss for consistency

**Review by CarsonKiibi — APPROVED** [review]
lgtm as mvp, need to update responsiveness (as described in pr) and dark mode in future.

**Review by IcedTea2K — APPROVED** [review]
Approve as per Carson's review

**Pull request merged**